### PR TITLE
Feature/add following

### PR DESCRIPTION
**Pull request by AndrewKaenko**

### CODE REVIEW

File: src/repositories/following.repository.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { FollowingModel } from "./../models/Following.model";
+
+export const getList = async (queryParams: object) => {
+  const query = {};
+  const options = {
+    ...queryParams,
+  };
+  const result = await FollowingModel.paginate(query, options);
+  return result;
+};
```
Comment (VitaliiDenysiuk) [inline src/repositories/following.repository.ts]:
```suggestion
export const getList = async (query: object, options: object) => {
  return FollowingModel.paginate(query, options);
};
```
check tweet repositories

File: src/routes/following.route.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  addFollowing,
+  deleteFollowing,
+} from "../services/following.service";
+
+const followingRouter = express.Router();
+
+followingRouter
+  .route("/")
+  .get(isAuthorized, async (req: Request, res: Response) => {
+    const result = await getFollowing(req.query);
```
Comment (VitaliiDenysiuk) [inline src/routes/following.route.ts]:
```suggestion
  .get(isAuthorized, async (req: Request, res: Response) => {
    const result = await getFollowing(new PaginationParameters(req));
```
 tweet router for example

File: src/routes/following.route.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    res.send(result);
+  });
+
+followingRouter
+  .route("/")
+  .post(isAuthorized, async (req: Request, res: Response) => {
+    await addFollowing(req.body);
+    res.sendStatus(200);
```
Comment (VitaliiDenysiuk) [inline src/routes/following.route.ts]:
```suggestion
    const newFollowing = await addFollowing(req.body);
    res.status(201).send(newFollowing);
```
we need to return created object

File: src/services/following.service.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import {
+  getList,
+  addOne,
+  deleteById,
+} from "../repositories/following.repository";
+
+export const getFollowing = async (queryParams: object) => {
+  const result = await getList(queryParams);
+  return result;
+};
```
Comment (VitaliiDenysiuk) [inline src/services/following.service.ts]:
```suggestion
export const getFollowing =  (queryParams: PaginationParameters<never, never>) => {
  return getList(...queryParams.get());
};
```
example in tweet service

File: src/services/following.service.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+export const addFollowing = async (data: Record<string, string>) => {
```
Comment (VitaliiDenysiuk) [inline src/services/following.service.ts]:
```suggestion
export const addFollowing =  (data: Record<string, string>) => {
```
why did you use async without await?

File: src/services/following.service.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  const { ownerId, followingId } = data;
+  return addOne(ownerId, followingId);
+};
+
+export const deleteFollowing = async (id: string) => {
```
Comment (VitaliiDenysiuk) [inline src/services/following.service.ts]:
```suggestion
export const deleteFollowing =  (id: string) => {
```
why did you use async without await?

**Review by VitaliiDenysiuk — CHANGES_REQUESTED** [review]

File: src/models/Following.model.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import mongoose, { Schema, Document } from "mongoose";
+import paginate from "mongoose-paginate-v2";
+
+export interface Following extends Document {
+  followingId: mongoose.Schema.Types.ObjectId;
+  ownerId: mongoose.Schema.Types.ObjectId;
```
Comment (VitaliiDenysiuk) [inline src/models/Following.model.ts]:
```suggestion
  followingId: Types.ObjectId;
  ownerId: Types.ObjectId;
```
https://mongoosejs.com/docs/typescript.html

File: src/models/Following.model.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  createdAt: string;
+  updatedAt?: string;
+}
+
+const FollowingSchema: Schema = new Schema<Following>({
+  followingId: {
+    type: mongoose.Schema.Types.ObjectId,
```
Comment (VitaliiDenysiuk) [inline src/models/Following.model.ts]:
```suggestion
    type: Schema.Types.ObjectId,
```
https://mongoosejs.com/docs/typescript.html

File: src/models/Following.model.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    required: true,
+    ref: "User",
+  },
+
+  ownerId: {
+    type: mongoose.Schema.Types.ObjectId,
```
Comment (VitaliiDenysiuk) [inline src/models/Following.model.ts]:
```suggestion
    type: Schema.Types.ObjectId,
```
https://mongoosejs.com/docs/typescript.html

File: src/models/Following.model.ts [hunk 10]
```diff
@@ -0,0 +1,40 @@
+import mongoose, { Schema, Document } from "mongoose";
+import paginate from "mongoose-paginate-v2";
+
+export interface Following extends Document {
+  followingId: mongoose.Schema.Types.ObjectId;
+  ownerId: mongoose.Schema.Types.ObjectId;
+  createdAt: string;
```
Comment (VitaliiDenysiuk) [inline src/models/Following.model.ts]:
```suggestion
  createdAt?: string;
```
we filled it by default

**Review by VitaliiDenysiuk — CHANGES_REQUESTED** [review]

**Review by VitaliiDenysiuk — APPROVED** [review]
looks good to me

**Review by alexandrtovmach — APPROVED** [review]

**Pull request merged**